server/webdriver/tests/vis_map_test.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import urllib
+import urllib.request
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from server.webdriver.base import WebdriverBaseTest
+import server.webdriver.shared as shared
+
+MAP_URL = '/tools/visualization#visType=map'
+URL_HASH_1 = '&place=geoId/06&placeType=County&sv=%7B"dcid"%3A"Count_Person_Female"%7D'
+PLACE_SEARCH_CA = 'California'
+
+
+# Class to test the map visualization tool.
+class TestVisMap(WebdriverBaseTest):
+
+  def test_server_and_page(self):
+    """Test the server can run successfully."""
+    self.driver.get(self.url_ + MAP_URL)
+
+    # Assert 200 HTTP code: successful page load.
+    req = urllib.request.Request(self.driver.current_url)
+    with urllib.request.urlopen(req) as response:
+      self.assertEqual(response.getcode(), 200)
+
+    # Assert 200 HTTP code: successful JS generation.
+    req = urllib.request.Request(self.url_ + '/visualization.js')
+    with urllib.request.urlopen(req) as response:
+      self.assertEqual(response.getcode(), 200)
+
+    # Assert page title is correct.
+    TITLE_TEXT = "Tools - Data Commons"
+    WebDriverWait(self.driver,
+                  self.TIMEOUT_SEC).until(EC.title_contains(TITLE_TEXT))
+    self.assertEqual(TITLE_TEXT, self.driver.title)
+
+    # Wait until the page has loaded.
+    element_present = EC.presence_of_element_located(
+        (By.CLASS_NAME, 'visualization-app'))
+    WebDriverWait(self.driver, self.TIMEOUT_SEC).until(element_present)
+
+    # Assert page heading and selected tab are correct.
+    page_header = self.driver.find_element(
+        By.XPATH, '//*[@id="main-pane"]/div/div[2]/div[1]/div[1]/h3')
```

Comment:
This xpath looks potentially brittle- any change to the dom structure could cause it to break. Instead, what do you think about selecting the relevant area by class name and seeing if it contains your text?

Maybe instead an expression like `'//*[@id="main-pane"]//*[contains(@class, "info-content")]'` (or a `CSS_SELECTOR`) and asserting that `'MapExplorer' in info_text.text`

Reply:
done